app/models/miq_request_workflow.rb

```diff
@@ -1075,11 +1075,8 @@ def allowed_storages(_options = {})
 
     rails_logger('allowed_storages', 0)
     st = Time.now
-    MiqPreloader.preload(hosts, :storages => {}, :host_storages => :storage)
 
-    storages = hosts.each_with_object({}) do |host, hash|
-      host.writable_accessible_storages.each { |s| hash[s.id] = s }
-    end.values
+    storages = Storage.where(:id => HostStorage.writable_accessible.where(:host => hosts).select(:storage_id))
     selected_storage_profile_id = get_value(@values[:placement_storage_profile])
     if selected_storage_profile_id
       storages.reject! { |s| !s.storage_profiles.pluck(:id).include?(selected_storage_profile_id) }
```

Comment:
@bdunne I just noticed this link, and I thikn there's an n+1 where when it tries to pull a storage profile for each storage.  I would expect we should preload storage_files before looping them (or find a way to reject them directly in the database.

Note that your changes did not affect this list and it was present previously.

Reply:
Also this can much more simply be written

```suggestion
      storages.reject! { |s| !s.storage_profile_ids.include?(selected_storage_profile_id) }
```

Reply:
cc @kbrock